### PR TITLE
Component page

### PR DESCRIPTION
**Pull request by sei1122**
added component page and updated navigation behaviour

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -313,6 +311,68 @@
       route: "/typography",
       component: <UXWTypography />,
     },
+    {
+      heading: "Component",
+      subheadings: [
+        {
+          subtitle: "Getting Started",
```
Comment (kaisenl) [inline main.js:318]: You don't need a subsection here if you don't need a link on the left.

File: main.js [hunk 2]
```diff
@@ -501,8 +561,7 @@
   }
 
   function NextSectionLink(props) {
-    if (props.index >= dswContent.length - 1)
-    {
+    if (props.index >= dswContent.length - 1) {
```
Comment (kaisenl) [inline main.js:564]: You might want to just remove the `NextSectionLink`s until they're fixed. They don't set the `activeMenuState` right now.

File: main.js [hunk 3]
```diff
@@ -542,32 +603,295 @@ function ScrollToTop() {
...
+                      reduxStore.dispatch(clearNavAction())
+                      reduxStore.dispatch(addNavAction("card"))
+                    }}>
+                      <ListItem button>
+                        <ListItemText disableTypography primary="Card" className={`pl-12 ${activeSubsections.findIndex((element) => element === "card") !== -1 ? classesMain.navitemselected : classesMain.navitem}`} />
+                      </ListItem>
+                    </RouterLink>
+                  }
+                </React.Fragment>
+                {/* <React.Fragment key="navlink-fragment-6">
```
Comment (kaisenl) [inline main.js:875]: Can probably just delete this instead of having it commented in here.

File: main.js [hunk 4]
```diff
@@ -576,10 +900,10 @@ function ScrollToTop() {
                 {dswContent.map(({ route, component, heading }, index) => {
                   return (
                     <Route key={`route-${index}`} path={route}>
-                      <div className="flex-col flex-shrink-0 pb-8" style={{fontFamily: "Overpass", color: "#152145"}} key={`section-${index}`}>
+                      <div className="flex-col flex-shrink-0 pb-8" style={{ fontFamily: "Overpass", color: "#152145" }} key={`section-${index}`}>
                         <h1 className="pb-4 uxw-text-display-2">{heading}</h1>
                         {component}
-                        <NextSectionLink index={index}/>
+                        <NextSectionLink index={index} />
```
Comment (kaisenl) [inline main.js:906]: I think you can remove this line to remove the Next Section link

File: UXWButton.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    <div className="container">
+      <div className="flex flex-row">
+        <div className="flex flex-col">
+          <p className="pb-4 max-w-prose">
+            Buttons indicates an action therefore text or icon + text on the button expresses what action will occur when you press it.
+          </p>
+        </div>
+        <div className="flex-grow"></div>
+      </div>
+      {/* <UXWSubsection anchor="button" title="Primary"> */}
```
[the remark this answers is not in the document]
Comment (sei1122) [reply UXWButton.js]: Updated

File: UXWCard.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+      <div className="flex flex-row">
+        <div className="flex flex-col">
+          <p className="pb-4 max-w-prose">
+            Each card contains information of a category.<br/>
+            Card contains a variety of content types like text, image and multimedia or button and links.
+          </p>
+        </div>
+        <div className="flex-grow"></div>
+      </div>
+      {/* <UXWSubsection anchor="#" title="Card"> */}
```
[the remark this answers is not in the document]
Comment (sei1122) [reply UXWCard.js]: Updated

File: UXWComponent.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    <div className="container">
+      <div className="flex flex-row">
+        <div className="flex flex-col">
+          <p className="pb-4 max-w-prose">
+          Components are reusable blocks of code. Components helps building UXW’s products efficiently and consistancy. 
+          </p>
+        </div>
+        <div className="flex-grow"></div>
+      </div>
+      <UXWSubsection anchor="getting-started" title="Getting Started">
```
[the remark this answers is not in the document]
Comment (sei1122) [reply UXWComponent.js]: Updated

File: index.html [hunk 8]
```diff
@@ -21,7 +21,7 @@
 
     <script crossorigin src="https://unpkg.com/babel-standalone@6/babel.min.js"></script>
 
-    <!-- Developer Links
+    <!--
```
[the remark this answers is not in the document]
Comment (sei1122) [reply index.html:24]: I'm not sure I understand correctly but when I revert, it doesn't render.

File: index.html [hunk 9]
```diff
@@ -43,7 +43,7 @@
     <script crossorigin src="https://unpkg.com/@reduxjs/toolkit@1.5.0/dist/redux-toolkit.umd.min.js"></script>
     <script crossorigin src="https://unpkg.com/react-redux@7.2.0/dist/react-redux.min.js"></script>
     <script crossorigin src="https://unpkg.com/@oopscurity/react-intersection-observer@6.4.0-alpha.1/dist/react-intersection-observer.umd.min.js"></script>
-
+
```
[the remark this answers is not in the document]
Comment (sei1122) [reply index.html:46]: I'm not sure I understand correctly but when I revert, it doesn't render.

File: main.js (hunk 2 above)
Comment (sei1122) [reply main.js:558]: When removing this section, it doesn't render.

File: main.js (hunk 3 above)
Comment (sei1122) [reply main.js]: done

File: main.js (hunk 4 above)
Comment (sei1122) [reply main.js]: done

File: main.js (hunk 1 above)
Comment (sei1122) [reply main.js]: done

File: UXWColorClass.js [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+    <div className="container">
+      <div className="flex flex-row">
+        <div className="flex flex-col">
+          <p className="pb-4 max-w-prose">
+            Brand colors are part of uxwcore.css. Use the color’s class name to update a HTML elements color.
+          </p>
+        </div>
+        <div className="flex-grow"></div>
+      </div>
+      {/* <UXWSubsection anchor="color-class" title="Background Color"> */}
```
Comment (kaisenl) [inline UXWColorClass.js:12]: This one too

File: UXWColorClass.js [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+.uxw-bg-acid-2 {
+  background-color: var(--uxw-color-merlins-beard-3);
+}`}
+              </div>
+            </UXWCodePanelHTMLCSS>
+          </div>
+        </div>
+        <p className="py-10">
+        CSS Color variables is set in 
+          <a href="https://github.com/UX-Wizards/design-system-web/blob/main/uxwcore.css" target="_blank">
```
Comment (kaisenl) [inline UXWColorClass.js:38]: image
You might need a `&nbsp;` or `{ }`

File: main.js [hunk 12]
```diff
@@ -313,6 +311,62 @@
       route: "/typography",
       component: <UXWTypography />,
     },
+    {
+      heading: "Component",
+      subheadings: [
+        {
```
Comment (kaisenl) [inline main.js:317]: You should be able to delete this empty entry. You don't have a subtitle or anchor anyway.

File: UXWComponent.js [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+        </div>
+        <div className="flex-grow"></div>
+      </div>
+      <p className="pb-4 max-w-prose">
+        To use this components, you need to include <span className="uxw-code">uxwcore.css</span> and <span className="uxw-code">uxw.css.</span>
+        Read more about <RouterLink to="/introduction" target="_blank">Installation and Usage </RouterLink>to learn how this works.
+      </p>
+      <div className="flex max-w-prose p-4 my-4 uxw-bg-steel-2 rounded-lg border-black rounded-md shadow-inner"><p className="max-w-prose">Use CSS class names on your HTML elements to apply the style. You don’t need to write CSS once you included <span className="uxw-code">uxwcore.css </span> and <span className="uxw-code">uxw.css</span> on your file.</p></div>
+      <div className="max-w-7xl grid grid-cols-3 gap-8 my-10">
+        <RouterLink to="/colorclass">
```
Comment (kaisenl) [inline UXWComponent.js:18]: When you click this, the left side probably won't be set properly. You'll probably need to pass the set function into UXWComponent, but I wouldn't worry about that for now.

File: UXWColorClass.js (hunk 10 above)
Comment (sei1122) [reply UXWColorClass.js]: updated

File: UXWColorClass.js (hunk 11 above)
Comment (sei1122) [reply UXWColorClass.js:37]: updated

File: main.js (hunk 12 above)
Comment (sei1122) [reply main.js:317]: updated

File: UXWComponent.js (hunk 13 above)
Comment (sei1122) [reply UXWComponent.js:18]: Good catch. I will write it down for future fixing

**Review by kaisenl — APPROVED** [review]
Ok cool, hopefully Max doesn't have a hard time merging in

**Pull request merged**